Helper/ClearLog.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Mageplaza
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Mageplaza.com license that is
+ * available through the world-wide-web at this URL:
+ * https://www.mageplaza.com/LICENSE.txt
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this extension to newer
+ * version in the future.
+ *
+ * @category    Mageplaza
+ * @package     Mageplaza_Smtp
+ * @copyright   Copyright (c) Mageplaza (https://www.mageplaza.com/)
+ * @license     https://www.mageplaza.com/LICENSE.txt
+ */
+
+namespace Mageplaza\Smtp\Helper;
+
+use Exception;
+use Magento\Framework\Stdlib\DateTime\DateTime;
+use Mageplaza\Smtp\Helper\Data;
+use Magento\Framework\App\ResourceConnection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class ClearLog
+ * @package Mageplaza\Smtp\Helper
+ */
+class ClearLog
+{
+    const LOG_TABLE = 'mageplaza_smtp_log';
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var Data
+     */
+    protected $helper;
+
+    /**
+     * @var DateTime
+     */
+    protected $date;
+
+    /**
+     * DB connection.
+     *
+     * @var ResourceConnection
+     */
+    protected $connection;
+
+    /**
+     * ClearLog constructor.
+     *
+     * @param LoggerInterface $logger
+     * @param DateTime $date
+     * @param ResourceConnection $connection
+     * @param Data $helper
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        DateTime $date,
+        ResourceConnection $connection,
+        Data $helper
+    )
+    {
+        $this->logger = $logger;
+        $this->date = $date;
+        $this->connection = $connection;
+        $this->helper = $helper;
+    }
+
+    /**
+     * Clean Email Log after X day(s)
+     *
+     * @return $this
+     */
+    public function execute()
+    {
+        if (!$this->helper->isEnabled()) {
```

Comment:
This check should be done in command console and in cron.
This helper should do the delete. On good example is that the console command could have a parameter that will overwrite the settings from database.
For example, the delete cron is disabled, but you should be capable to clean the table, if you want.